### ISSUE TITLE
Returning calculations in form of default units?

### ISSUE THREAD

**Issue by kaylai** [kaylai/VESIcal#135]
I assumed that passing set_default_units() on some sample meant that any calculations returned would be in the default units set. Was this the intention? I've coded it up so that MagmaSat does this (e.g., if default_units are set to "mol_oxides", then calculate_dissolved_volatiles() returns dissolved H2O and CO2 values in mol fraction). I also coded up the BatchModel class to make this work, but it does not seem to work when I tested it on the Dixon model.

Perhaps this was not the intended use?

Example code:
```python
import VESIcal as v

mysample = v.Sample({'SiO2':    47.95,
                                 'TiO2':    1.67,
                                 'Al2O3':   17.32,
                                 'FeO':     10.24,
                                 'Fe2O3':   0.1,
                                 'MgO':     5.76,
                                 'CaO':     10.93,
                                 'Na2O':    3.45,
                                 'K2O':     1.99,
                                 'P2O5':    0.51,
                                 'MnO':     0.1})

mysample.set_default_units("mol_oxides")
disvols = v.calculate_dissolved_volatiles(mysample, temperature=1000, pressure=1000, X_fluid=1, model="Dixon").result
```

This produces the result disvols = {'H2O_liq': 3.033261541386283, 'CO2_liq': 0.0}. Clearly in wt% not mole fraction.

If the default units are mol_oxides, do we want this to return as mol_oxides? I think yes, unless there is a reason not to. Thoughts?

**Comment by simonwmatthews** [comment]
You're right- `using set_default_units()` does mean the default return from `get_composition()` will be in those units. However, if particular solubility models have to use a particular unit or particular normalisation, then the default units and normalisation will be ignored, these retrieve the composition by using something like `get_composition(units='wtpt_oxides', normalization='standard')`. 

I've not built in any logic within the models to lookup the default units for a sample and alter their output accordingly, though I like this idea. I had no real intention for how the customisable default units should be used, I just thought it would make sense to add this as an option.

**Comment by kaylai** [comment]
@simonwmatthews That makes total sense. I did it with MagmaSat by ensuring the composition for the calculations was in the units required (wt%) and then at the end of the calculation translating back into the default units set for that Sample or BatchFile object. Let me take a crack at it with the Dixon model and see what I can come up with.

**Comment by kaylai** [comment]
Ok, I think this is all working now! I made some tests of my own to verify that the results are correct for all models, both for single sample and batchfile calculations. I implemented this by adding a method to the calculate_dissolved_volatiles() calculate class that translates the results of a calculation into the passed object's default units.

I started making the unit tests for this but will finalize those later.
